src/ServiceBusSQSChannel.php

```diff
@@ -50,21 +59,61 @@ public function send($notifiable, Notification $notification)
             return;
         }
 
-        $message = $notification
-            ->toServiceBus($notifiable)
-            ->getParams();
+        if (!isset($params['from'], $params['events'][0])) {
+            Log::error('Invalid message structure', ['params' => $params]);
+            return;
+        }
+
+        $queueUrl = Arr::get($this->config, 'sqs.queue_url');
+        $isFifoQueue = strpos($queueUrl, '.fifo') !== false;
 
-        $response = $this->sqs->sendMessage([
-            'QueueUrl' => Arr::get($this->config, 'sqs.queue_url'),
-            'MessageBody' => json_encode($message),
-            'MessageGroupId' => $message['from'],
-        ]);
+        $payloadSqs = [
+            'QueueUrl' => $queueUrl,
+            'MessageBody' => json_encode($params),
+        ];
 
-        $event = $message['events'][0];
+        if ($isFifoQueue) {
+            $payloadSqs['MessageGroupId'] = $params['from'];
+            $payloadSqs['MessageDeduplicationId'] = md5(json_encode($params));
+        }
 
-        Log::info("{$event} sent to bus queue", [
-            'message_id' => $response->get('MessageId'),
-            'message' => $message,
-        ]);
+        try {
+            $response = $this->sqs->sendMessage($payloadSqs);
+
+            $eventName = $params['events'][0];
+
+            Log::info("{$eventName} sent to bus queue", [
+                'message_id' => $response->get('MessageId'),
+                'params' => $params,
+            ]);
+
+            $this->hasAttemptedRefresh = false;
+        } catch (AwsException $exception) {
+            $code = $exception->getAwsErrorCode();
+
+            if (in_array($code, ['ExpiredToken', 'UnrecognizedClientException', 'InvalidClientTokenId'])) {
+                Log::info("$code received. Refreshing credentials and retrying.", [
+                    'event' => $eventType,
+                    'params' => $params,
+                    'aws_error_code' => $code,
+                    'aws_error_message' => $exception->getAwsErrorMessage(),
+                    'tags' => ['service-bus'],
+                ]);
+
+                if (!$this->hasAttemptedRefresh) {
+                    $this->hasAttemptedRefresh = true;
+
+                    $this->initializeSqsClient();
+
+                    $this->send($notifiable, $notification);
```

Comment:
We miss the log bit. i suggest we place the `sendMessage` and `log` in one call so we can reuse here and in the try for consistency.